python/plugins/grassprovider/ext/r_horizon.py

```diff
@@ -38,14 +39,42 @@ def checkParameterValuesBeforeExecuting(alg, parameters, context):
 
 
 def processOutputs(alg, parameters, context, feedback):
+    # Inspired from GRASS implementation
+    def getNumberDecimals(number):
+        """ Return the number of decimals of a number (int or float) """
+        if int(number) == number:
+            return 0
+        return len(str(number).split(".")[-1])
+
+    def doubleToBaseName(number, nDecimals):
+        """
+        Format filename, according to GRASS implementation,
+        based on provided number and number of decimals
+        """
+        number += 0.0001
```

Comment:
Why 0,0001? Is this still fixed? Is it to avoid rounding errors later?

Reply:
A comment could be useful.

Reply:
@lbartoletti, this replicates the logic in https://github.com/OSGeo/grass/blob/6253da1bd6ce48d23419e99e8b503edf46178490/raster/r.horizon/main.c#L1156-L1170